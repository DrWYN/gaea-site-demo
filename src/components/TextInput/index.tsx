import * as React from 'react';

interface ITextInputProps {
  testID?: string;
  className?: string;
  style?: any;
  value?: string;
  autoCapitalize?: string;
  autoFocus?: boolean;
  blurOnSubmit?: boolean;
  defaultValue?: string;
  editable?: boolean;
  type?: string;
  maxLength?: number;
  multiline?: boolean;
  numberOfLines?: number;
  onBlur?: any;
  onChange?: any;
  onFocus?: any;
  onKeyDown?: any;
  onKeyPress?: any;
  onSelectionChange?: any;
  placeholder?: string;
  secureTextEntry?: boolean;
}

class TextInput extends React.Component<ITextInputProps, any> {
  static displayName = 'TextInput'; 

  render() {
    const {
      multiline,
      testID,
      numberOfLines=2,
      editable=true,
      ...other 
    } = this.props;

    if(multiline){
      return <textarea 
        dir={"auto"}
        rows={numberOfLines}
        readOnly={!editable}
        {...other}/>
    } else {
      return <input 
        dir={"auto"}
        readOnly={!editable}
        {...other}/>
    }
  }
}
export default TextInput;
