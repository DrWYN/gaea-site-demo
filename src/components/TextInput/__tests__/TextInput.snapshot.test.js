import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../index';

describe('<TextInput />', () => {

  it('no props', () => {
    const component = renderer.create(<TextInput />);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('multiline', () => {
    const component = renderer.create(<TextInput multiline={true} numberOfLines={100} editable={true}/>);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
})
