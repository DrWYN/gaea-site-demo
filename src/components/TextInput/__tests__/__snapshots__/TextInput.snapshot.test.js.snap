// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> multiline 1`] = `
<textarea
  dir="auto"
  readOnly={false}
  rows={100}
/>
`;

exports[`<TextInput /> no props 1`] = `
<input
  dir="auto"
  readOnly={false}
/>
`;
